Terminal.Gui/Application.cs

```diff
@@ -1183,38 +1190,75 @@ public static void OnMouseEvent (MouseEventEventArgs a)
 				}
 			}
 
-			if (view != null) {
-				var nme = new MouseEvent () {
-					X = rx,
-					Y = ry,
-					Flags = a.MouseEvent.Flags,
-					OfX = 0,
-					OfY = 0,
-					View = view
-				};
+			bool FrameHandledMouseEvent (Frame frame)
+			{
+				if (frame == null) {
+					return false;
+				}
 
-				if (_lastMouseOwnerView == null) {
-					_lastMouseOwnerView = view;
-					view.OnMouseEnter (nme);
-				} else if (_lastMouseOwnerView != view) {
-					_lastMouseOwnerView.OnMouseLeave (nme);
-					view.OnMouseEnter (nme);
-					_lastMouseOwnerView = view;
+				var boundsPoint = frame.ScreenToBounds (a.MouseEvent.X, a.MouseEvent.Y);
+
+				if (frame?.Thickness.Contains (boundsPoint.X, boundsPoint.Y) ?? false) {
+					var me = new MouseEvent () {
+						X = boundsPoint.X,
+						Y = boundsPoint.Y,
+						Flags = a.MouseEvent.Flags,
+						OfX = boundsPoint.X,
+						OfY = boundsPoint.Y,
+						View = frame
+					};
+					if (frame.OnMouseEvent (me)) {
+						return true;
+					}
 				}
+				return false;
+			}
 
-				if (!view.WantMousePositionReports && a.MouseEvent.Flags == MouseFlags.ReportMousePosition)
-					return;
+			// Work inside-out (Padding, Border, Margin)
+			// TODO: Debate whether inside-out or outside-in is the right strategy
```

Comment:
Inside out would make the most sense I think.  That would match the way the rest of the mouse events work with Views i.e. the 'most focused' View gets first bite then key goes up the nesting chain.

So:

- View handle mouse event
- Padding handle mouse event
- Frame handle mouse event
- Margin handle mouse event

Reply:
In my design, view ONLY gets mouse events that occur within the View. So even if Margin/Border/Padding handle the mouse event (or not) the view will never see it (because, by definition, the coords are outside of Bounds.

So the order really doesn't matter.